lib/game.rb

```diff
@@ -120,42 +124,22 @@ def switch_player
         @computer_player
     end
 
-    def win?
-      @players.each do |player|
-        if is_winner?(player)
-          return true
+    def declare_outcome(board, markers)
+      if TicTacToe::is_game_over?(board, markers) 
+        if TicTacToe::draw?(board, markers)
+          return DRAW
+        elsif TicTacToe::win?(board, markers)
+          determine_winner
         end
       end
-
-      false
-    end
-
-    def draw?
-      @board.is_grid_filled? && win? == false
     end
 
-    def is_winner?(player)
-      if @board.is_either_diagonal_filled?(player.marker)
-        @winner = player
-        return true
-      end
-
-      @board.size.times do |n| 
-        if @board.is_row_filled?(n, player.marker) || @board.is_column_filled?(n, player.marker)
-          @winner = player
-          return true
+    def determine_winner
```

Comment:
Similar here from my previous comment - `GameState` could take care of this implementation (or maybe even another module!)